src/test/txpackage_tests.cpp

```diff
@@ -0,0 +1,566 @@
+        CMutableTransaction mtx_parent = CreateValidTransaction(spend_coinbase, CAmount(49 * COIN), {CScript() << OP_TRUE});
+        Sign(vchSig, coinbaseKey, m_coinbase_txns[index_cb]->vout[0].scriptPubKey, 0, mtx_parent, 0);
+        mtx_parent.vin[0].scriptSig = CScript() << vchSig;
+        package_unrelated.emplace_back(MakeTransactionRef(mtx_parent));
+
+    }
+    submitPool.clear();
+    auto result_unrelated = TestPackageAcceptance(package_unrelated, state, packageState, opt);
+
+    // We don't expect m_tx_results for each transaction when basic sanity checks haven't passed.
+    BOOST_CHECK(result_unrelated);
+    BOOST_CHECK(state.IsValid());
+    BOOST_CHECK_EQUAL(state.GetRejectCode(), 0);
+    BOOST_CHECK_EQUAL(state.GetRejectReason(), "");
+    for(auto& s : packageState) {
+        BOOST_CHECK_EQUAL(s.second.GetRejectCode(), 0);
+        BOOST_CHECK_EQUAL(s.second.GetRejectReason(), "");
+    }
+    BOOST_CHECK_EQUAL(submitPool.size(), 10);
+
+    auto result_unrelated_submit = SubmitPackageToMempool(submitPool, stateSubmit);
+
+    BOOST_CHECK(result_unrelated_submit);
+    BOOST_CHECK_EQUAL(stateSubmit.GetRejectCode(), 0);
+    BOOST_CHECK_EQUAL(stateSubmit.GetRejectReason(), "");
+    BOOST_CHECK_EQUAL(mempool.size(), expected_pool_size + 10);
+
+    // Parent and Child (and Grandchild) Package
+    ++index_cb;
+    Package package_parent_child;
+    Package package_3gen;
+    COutPoint spend_cbase(m_coinbase_txns[index_cb]->GetHashMalFix(), 0);
+    CMutableTransaction mtx_parent = CreateValidTransaction(spend_cbase, CAmount(49 * COIN), parent_locking_script);
+    Sign(vchSig, coinbaseKey, m_coinbase_txns[index_cb]->vout[0].scriptPubKey, 0, mtx_parent, 0);
+    mtx_parent.vin[0].scriptSig = CScript() << vchSig;
+    CTransactionRef tx_parent{MakeTransactionRef(mtx_parent)};
+    package_parent_child.push_back(tx_parent);
+    package_3gen.push_back(tx_parent);
+
+    CKey child_key;
+    child_key.MakeNewKey(true);
+    CScript child_locking_script = GetScriptForDestination(child_key.GetPubKey().GetID());
+    COutPoint spend_parent(tx_parent->GetHashMalFix(), 0);
+    CMutableTransaction mtx_child = CreateValidMempoolTransaction(mempool, spend_parent, CAmount(48 * COIN), child_locking_script);
+    Sign(vchSig, coinbaseKey, m_coinbase_txns[index_cb]->vout[0].scriptPubKey, 0, mtx_parent, 0);
+    mtx_parent.vin[0].scriptSig = CScript() << vchSig;
+    CTransactionRef tx_child{MakeTransactionRef(mtx_child)};
+    package_parent_child.push_back(tx_child);
+    package_3gen.push_back(tx_child);
+
+    COutPoint spend_child(tx_child->GetHashMalFix(), 0);
+    CMutableTransaction mtx_grandchild = CreateValidMempoolTransaction(mempool, spend_child, CAmount(47 * COIN), CScript() << OP_TRUE << OP_EQUAL );
+    mtx_grandchild.vin[0].scriptSig = CScript() << OP_TRUE;
+    CTransactionRef tx_grandchild{MakeTransactionRef(mtx_grandchild)};
+    package_3gen.push_back(tx_grandchild);
+
+    // 3 Generations is allowed.
```

Comment:
In Bitcoin, this is not allowed.

Reply:
Yes. In Tapyrus we do not differentiate "package" and "child with unconfirmed parent" as different types of packages. So this is accepted. Bitcoin does not accept such a package in the second type of package only. I think this purpose of the second category is fee related, to allow 'child pays for parent' packages.